frappe/core/doctype/file/utils.py

```diff
@@ -234,7 +235,12 @@ def _save_file(match):
 			content = content.encode("utf-8")
 		if b"," in content:
 			content = content.split(b",")[1]
-		content = safe_b64decode(content)
+
+		try:
+			content = safe_b64decode(content)
+		except BinasciiError:
+			frappe.flags.has_dataurl = True
+			return '<img src=""'
```

Comment:
I think we should just return the "match" instead. We failed to parse the data, what we can just leave this at is maintain the content as is. And that should show the "broken image" icon we universally love and understand right?

If this makes sense, you can disregard the following comment

Reply:
Returning the original content does not work because it would be removed by `sanitize_html` downstream. Now we're returning `<img src="#broken-image"` to get the "broken image" icon.

Reply:
Sounds good 👍🏼 

Additionally, I wondered how we could add some more context to this when a user is confused by the broken image - Adding a title attribute to the tag saying something like "Image: Corrupted Data Stream" could make scenarios like that more...debug-able

Reply:
Done, @gavindsouza

Reply:
Can confirm this is better 👍🏼